src/Aperture/overlap.jl

```diff
@@ -154,3 +154,117 @@ function area_arc(x0, y0, x1, y1, r)
+"""
+function point_completely_inside_ellipse(x, y, h, k, cxx, cyy, cxy)
+    if cxx * (x - h)^2 + cxy *(x - h)*(y - k) + cyy * (y - k)^2  - 1 > 0
+        return false
+    else
+        return true
+    end
+end
+
+
+function elliptical_overlap(xmin, xmax, ymin, ymax, nx, ny, a, b, theta; method = :center)
+    out = fill(0.0, nx, ny)
+
+    # width of each element
+    dx = (xmax - xmin) / nx
+    dy = (ymax - ymin) / ny
+
+    # radius of one pixel
+    pixel_radius = 0.5sqrt(dx^2 + dy^2)
+
+    # bounding box
+    bxmin, bxmax, bymin, bymax = bbox(EllipticalAperture(0, 0, a, b, theta))
+
+    bxmin -= 0.5dx
+    bxmax += 0.5dx
+    bymin -= 0.5dy
+    bymax += 0.5dy
+
+    cxx, cyy, cxy = oblique_coefficients(a, b, theta)
+
+    for i in 1:nx
+        # lower end of pixel
+        pxmin = xmin + (i - 1) * dx
+        pxcen = pxmin + 0.5dx
+        # upper end of pixel
+        pxmax = pxmin + dx
+
+        if pxmax > bxmin && pxmin < bxmax
+            for j in 1:ny
+                pymin = ymin + (j - 1) * dy
+                pycen = pymin + 0.5dy
+                pymax = pymin + dy
+                if pymax > bymin && pymin < bymax
+
+                    #4 flags for four different ends of the pixel
+                    #each flag tells wether the point is inside the ellipse or not
+                    #sample strcture of a pixel is shown below
+                    #  2---4
+                    #  |   |
+                    #  |   |
+                    #  1---3
+
+                    flag1 = point_completely_inside_ellipse(pxmin, pymin, 0, 0, cxx, cyy, cxy)
+                    flag2 = point_completely_inside_ellipse(pxmin, pymax, 0, 0, cxx, cyy, cxy)
+                    flag3 = point_completely_inside_ellipse(pxmax, pymin, 0, 0, cxx, cyy, cxy)
+                    flag4 = point_completely_inside_ellipse(pxmax, pymax, 0, 0, cxx, cyy, cxy)
+
+                    # fully within radius
+                    if flag1 && flag2 && flag3 && flag4
+                        @inbounds out[j, i] = 1
+                    # partially within radius
+                    elseif flag1 || flag2 || flag3 || flag4
+                        if method === :exact
+                            print("This method has not yet been implemented!")
+                        elseif method === :center
+                            @inbounds out[j, i] =  elliptical_overlap_single_subpixel(pxmin, pymin, pxmax, pymax, cxx, cyy, cxy, 1)
+                        elseif method[1] === :subpixel
+                            @inbounds out[j, i] =  elliptical_overlap_single_subpixel(pxmin, pymin, pxmax, pymax, cxx, cyy, cxy, method[2])
+                        end
+                    end
+                end
+            end
+        end
+    end
+    return out
+end
+
+elliptical_overlap(xmin, xmax, ymin, ymax, nx, ny, parameters::AbstractVector; method) = elliptical_overlap(xmin, xmax, ymin, ymax, nx, ny, parameters... ; method = :center)
```

Comment:
I don't have any coverage info, but is this function ever used? If not, I would delete it.

Since all of these functions are internally used and not intended to be exported, we don't need to make anything other than we use.

Reply:
This function is being used during testing.